src/models/PenaltyIdentifierSchemaFactory.ts

```diff
@@ -0,0 +1,94 @@
+import Joi from '@hapi/joi';
+
+type PrefixVariation = { upperCase: string[]; lowerCase: string[]; };
+type PrefixMap = { singleCharacter: PrefixVariation, doubleCharacter: PrefixVariation; };
+
+export class PenaltyIdentifierSchemaFactory {
```

Comment:
The majority of the logic in this class is related to the company number, is it worth separating the two?

Reply:
I don't think it adds much value since throughout the project we tend to use the PenaltyIdentifier schema as a whole

Reply:
ok fair enough